### DISCUSSION TITLE
Do not use parent property in artifactId dependency for maven project

### DISCUSSION THREAD

**Discussion by djice** [renovatebot/renovate#24875] [Suggest an Idea]
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### Please select which platform you are using if self-hosting.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

With maven I have a dependency like
```xml
        <dependency>
            <groupId>io.circe</groupId>
            <!--  scala.binary.version property comes from parent   -->
            <artifactId>circe-parser_${scala.binary.version}</artifactId>
            <version>0.14.0</version>
        </dependency>
```

The property "${scala.binary.version} comes from parent.

The issue is that renovate does not create a merge request for this library.

If I force the property like that it works
```xml
        <dependency>
            <groupId>io.circe</groupId>
            <artifactId>circe-parser_2.13</artifactId>
            <version>0.14.0</version>
        </dependency>
```

I make an exemple project to illustrate the issue:

https://github.com/djice/renovate-issue

In this project https://github.com/djice/renovate-issue/blob/main/maven-property-from-parent/with-property-in-parent/pom.xml the merge request is not created

In this project https://github.com/djice/renovate-issue/blob/main/maven-property-from-parent/with-property-inside/pom.xml the merge request is well created here https://github.com/djice/renovate-issue/pull/3

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by gaeljw** [comment]
For the record, duplicated at https://github.com/renovatebot/renovate/discussions/21510
